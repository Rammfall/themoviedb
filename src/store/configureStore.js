import { createLogicMiddleware } from 'redux-logic'
import { createStore, applyMiddleware } from 'redux'

import logics from './logics'
import rootReducer from './rootReducer'

const bindMiddleware = (middleware) => {
  if (process.env.NODE_ENV !== 'production') {
    // eslint-disable-next-line
    const { composeWithDevTools } = require('redux-devtools-extension')

    return composeWithDevTools(applyMiddleware(...middleware))
  }

  return applyMiddleware(...middleware)
}

const configureStore = (initialState = {}) => {
  // Add dependencies to pass them to logic functions
  const dependencies = {}

  const logicMiddleware = createLogicMiddleware(logics, dependencies)
  const store = createStore(rootReducer, initialState, bindMiddleware([logicMiddleware]))

  return store
}

export default configureStore
