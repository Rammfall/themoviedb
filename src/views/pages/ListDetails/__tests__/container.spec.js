import React from 'react'
import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import diveTo from 'TestUtils/diveToEnzyme'

import ListDetailsPage, { ListDetailsPageContainer } from '../container'

jest.mock('react-router-dom', () => ({
  withRouter: component => component
}))

jest.mock('Store/concepts/listDetails/selectors', () => ({
  isBlankSelector: jest.fn(() => true)
}))

describe('ListDetailsPage', () => {
  const store = configureStore()()
  const props = {
    match: {
      params: {
        id: '1'
      }
    },
    location: {
      search: ''
    }
  }
  const wrapper = shallow(
    <ListDetailsPage
      store={store}
      {...props}
    />
  )
  const container = diveTo(wrapper, ListDetailsPageContainer)

  it('matches snapshot', () => {
    expect(container).toMatchSnapshot()
  })

  describe('componentDidUpdate()', () => {
    it('matches snapshot', () => {
      const spy = jest.spyOn(container.instance(), 'componentDidUpdate')
      container.setProps({ location: { search: '?page=3' } })
      container.setProps({ location: { search: '?page=3' } })

      expect(spy).toHaveBeenCalled()
    })
  })
})
