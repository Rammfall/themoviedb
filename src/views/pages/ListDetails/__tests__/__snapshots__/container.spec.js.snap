// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListDetailsPage matches snapshot 1`] = `
<ListDetails
  isBlank={true}
/>
`;
